calc.js

```diff
@@ -0,0 +1,502 @@
+      break;
+    case "base10":
+      screen.innerHTML = Math.pow(10, screen.innerHTML);
+      break;
+    case "logarithm":
+      if (screen.innerHTML === "") {
+        alert("Enter the number you want");
+      } else {
+        screen.innerHTML = Math.log(screen.innerHTML);
+      }
+      break;
+    case "natural-logarithm":
+      if (screen.innerHTML == "") {
+        alert("Enter the number you want");
+      } else {
+        screen.innerHTML = Math.log10(screen.innerHTML);
+      }
+      break;
+    case "factorial":
+      screen.innerHTML = factorialFunc(screen.innerHTML);
+      break;
+    case "memory-stored":
+      screen.innerHTML = memoryStored();
+      break;
+    case "memory-recall":
+      screen.innerHTML = memoryRecall();
+      break;
+    case "memory-clear":
+      memoryClear();
+      break;
+    case "memory-plus":
+      memoryPlus(screen.innerHTML);
+      break;
+    case "memory-minus":
+      memoryMinus();
+      break;
+    case "pow3":
+      if (screen.innerHTML.includes("-")) {
+        screen.innerHTML = -1 * Math.pow(-1 * screen.innerHTML, 3);
+      } else {
+        screen.innerHTML = Math.pow(screen.innerHTML, 3);
+      }
+      break;
+    case "qube-root":
+      if (screen.innerHTML.includes("-")) {
+        screen.innerHTML = -1 * Math.pow(-1 * screen.innerHTML, 1 / 3);
+      } else {
+        screen.innerHTML = Math.pow(screen.innerHTML, 1 / 3);
+      }
+      break;
+    case "x-root-y":
+      screen.innerHTML += "^";
+      break;
+    case "base-e":
+      screen.innerHTML = Math.pow(Math.E, screen.innerHTML);
+      break;
+    case "base2":
+      screen.innerHTML = Math.pow(2, screen.innerHTML);
+      break;
+    case "log2":
+      screen.innerHTML = Math.log2(screen.innerHTML);
+      break;
+  }
+}
+function backSpace(value) {
+  return value.substring(0, value.length - 1);
+}
+//deg and rad function
+degree.addEventListener("click", () => {
+  if (!degree.classList.contains("show")) {
+    degree.classList.add("show");
+    degree.innerHTML = "RAD";
+    Flag = false;
+  } else {
+    degree.classList.remove("show");
+    degree.innerHTML = "DEG";
+    Flag = true;
+  }
+});
+//Function of factorial
+function factorialFunc(n) {
+  for (let i = 1; i <= n; i++) {
+    if (n === 1 || n === 0) {
+      return 1;
+    } else {
+      return n * factorialFunc(n - 1);
+    }
+  }
+}
+//function of root
+function root(num) {
+  let a, b;
+  a = num.slice(0, num.indexOf("^"));
+  b = num.slice(num.indexOf("^") + 1);
+  return Math.pow(a, b);
+}
+//memory functions
+function memoryClear() {
+  for (let i = 0; i <= memory.length; i++) {
+    memory.pop();
+  }
+  console.log(memory);
+  return memory;
+}
+function memoryRecall() {
+  return memory.pop().toString();
+}
+function memoryPlus(num) {
+  if (memory.length === 0) {
+    screen.innerHTML = "";
+    memory.push(num);
+    console.log(memory);
+  } else {
+    memory.push(num);
+  }
+}
+function memoryMinus() {
```

Comment:
memory minus logic is not appropriate. this calculator function not remove number from memory. it save number in memory with negative value